### ISSUE TITLE
smtp4dev attempts to access the file system despite using in-memory database

### ISSUE THREAD

**Issue by kpwbo** [rnwood/smtp4dev#302]
In the BuildWebHost method of Program.cs, the code attempts to create a directory in %APPDATA%/smtp4dev, despite having specified ServerOptions.Database as empty (using in-memory database).

Is the directory used for anything other than storing the database? If not, the code shouldn't attempt to create a directory (or write anything on the file system) when we're using the in-memory database.

This is important because in our case smtp4dev does not have write access to the file system and we use the in-memory database, so the newest versions of smtp4dev fail on startup. We rolled back to 3.1.0-ci2020031904, which works correctly for us.

**Comment by billybull** [comment]
After running into an issue attempting to use docker image rnwood/smtp4dev:v3 I was getting error:

'System.UnauthorizedAccessException: Access to the path "C:\smtp4dev" is denied.'

I believe this issue was the reason. I tried using version 3.1.0-ci2020042101 without success (same error) and then finally found this open issue and tried 3.1.0-ci2020031904 as was suggested by #kpwbo, which appears to have resolved the error. Hope this helps. Thank you so much for your amazing work all!

**Comment by rnwood** [comment]
I agree that the app should not need write access to the install dir to operate.
Recent versions now put the DB and auto generated self-signed cert in %APPSETTINGS%. In this case I think it was the self signed cert generation causing the writes to the app dir.

**Comment by qoobic** [comment]
This issue appears to be present when running v3.1.2 on Docker. Output below:

```
PS C:\Users\Administrator> docker run rnwood/smtp4dev:windows-amd64-3.1.2
smtp4dev version 3.1.2-ci20210315103
https://github.com/rnwood/smtp4dev

.NET Core runtime version: 3.1.2


 > For help use argument --help

docker : Unhandled exception. System.UnauthorizedAccessException: Access to the path 'C:\smtp4dev' is denied.
At line:1 char:1
+ docker run rnwood/smtp4dev:windows-amd64-3.1.2
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Unhandled excep...dev' is denied.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
   at System.Environment.set_CurrentDirectoryCore(String value)
   at System.Environment.set_CurrentDirectory(String value)
   at System.IO.Directory.SetCurrentDirectory(String path)
   at Rnwood.Smtp4dev.Program.BuildWebHost(String[] args) in C:\app\Rnwood.Smtp4dev\Program.cs:line 96
   at Rnwood.Smtp4dev.Program.Main(String[] args) in C:\app\Rnwood.Smtp4dev\Program.cs:line 34
```

I've included the output from v3.1.0-ci2020031904 below for comparison:

```
PS C:\Users\Administrator> docker run rnwood/smtp4dev:windows-amd64-3.1.0-ci2020031904
smtp4dev version 3.1.0-ci2020031904
https://github.com/rnwood/smtp4dev

.NET Core runtime version: 3.1.2

Using Sqlite database at C:\app\database.db
SMTP Server is listening on port 25. Keeping last 100 messages and 100 sessions.
Hosting environment: Production
Content root path: C:\app
Now listening on: http://[::]:80
Application started. Press Ctrl+C to shut down.
Message/session consuming thread running.
```
